src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -207,53 +201,20 @@ ucs_status_t uct_ib_mlx5_get_cq(struct ibv_cq *cq, uct_ib_mlx5_cq_t *mlx5_cq)
         cqe->op_own |= MLX5_CQE_OWNER_MASK;
     }
 
-
     return UCS_OK;
 }
 
 static int
 uct_ib_mlx5_res_domain_cmp(uct_ib_mlx5_res_domain_t *res_domain,
                            uct_ib_md_t *md, uct_priv_worker_t *worker)
 {
-#ifdef HAVE_IBV_EXP_RES_DOMAIN
-    return res_domain->ibv_domain->context == md->dev.ibv_context;
-#elif HAVE_DECL_IBV_ALLOC_TD
     return res_domain->pd->context == md->dev.ibv_context;
-#else
-    return 1;
-#endif
 }
 
 static ucs_status_t
 uct_ib_mlx5_res_domain_init(uct_ib_mlx5_res_domain_t *res_domain,
                             uct_ib_md_t *md, uct_priv_worker_t *worker)
 {
-#ifdef HAVE_IBV_EXP_RES_DOMAIN
-    struct ibv_exp_res_domain_init_attr attr;
-
-    attr.comp_mask    = IBV_EXP_RES_DOMAIN_THREAD_MODEL |
-                        IBV_EXP_RES_DOMAIN_MSG_MODEL;
-    attr.msg_model    = IBV_EXP_MSG_LOW_LATENCY;
-
-    switch (worker->thread_mode) {
-    case UCS_THREAD_MODE_SINGLE:
-        attr.thread_model = IBV_EXP_THREAD_SINGLE;
-        break;
-    case UCS_THREAD_MODE_SERIALIZED:
-        attr.thread_model = IBV_EXP_THREAD_UNSAFE;
-        break;
-    default:
-        attr.thread_model = IBV_EXP_THREAD_SAFE;
-        break;
-    }
-
-    res_domain->ibv_domain = ibv_exp_create_res_domain(md->dev.ibv_context, &attr);
```

Comment:
the res_domain code seems broken: 
1. we should not really every have MULTI https://github.com/openucx/ucx/blob/2481433830d318be88652ce9ab7d5faf2511cafc/src/ucp/core/ucp_worker.c#L2262
2. seems we never really allocate res_domain; we just take it from md, but it can create a problem, can multiple workers share same UAR on the PD?